docs/pages/machine-id/guides/github-actions-kubernetes.mdx

```diff
@@ -0,0 +1,220 @@
+Kubernetes role to that group.
+
+This is the stage where you will actually determine what level of access the
+bot will have when accessing your cluster. In our example, we are applying the
+`view` ClusterRole which will allow our bot to read, but not write, all
+resources in the Kubernetes cluster.
+
+In your case you may wish to create a custom Kubernetes role that lists the
+exact permissions that you want your bot to have. You may also wish to use a
+RoleBinding rather than a ClusterRoleBinding to limit the bot's access to an
+individual Kubernetes namespace.
+
+Create a file named `clusterolebinding.yaml` and insert:
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: github-demo-group
+subjects:
+  - kind: Group
+    # Note that this name field matches the group name we provided in the
+    # Teleport role configuration.
+    name: github-demo
+    apiGroup: rbac.authorization.k8s.io
+roleRef:
+  kind: ClusterRole
+  # "view" is a default ClusterRole that grants read-only access to resources
+  # See: https://kubernetes.io/docs/reference/access-authn-authz/rbac/#user-facing-roles
+  name: view
+  apiGroup: rbac.authorization.k8s.io
+```
+
+Apply this to your cluster using `kubectl`:
+
+```
+$ kubectl apply -f ./clusterolebinding.yaml
+```
+
+## Step 4/4. Create the GitHub Actions workflow
+
+With all the configuration in Teleport and Kubernetes complete, you can now
+create a GitHub Actions workflow that can connect to your Kubernetes cluster.
+
+Our example workflow will list all of the pods contained within the cluster,
+but this could equally be modified to deploy to a Kubernetes cluster with
+`kubectl` or `helm`.
+
+In the GitHub workflows directory of your repository (`.github/workflows/`)
+create a new workflow YAML file, in this case `actionstest.yml`.
+
+Insert the following, ensuring that you replace `my-kubernetes-cluster` with the
+name of your Kubernetes cluster and `example.domain:443` with the address of
+your Teleport Proxy.
+
+```yaml
+# This is a basic workflow to help you get started, modify it for your needs.
+on:
+  push:
+    branches:
+      - main
+jobs:
+  demo:
+    permissions:
+      # The "id-token: write" permission is required or Machine ID will not be 
+      # able to authenticate with the cluster.
+      id-token: write
+      contents: read
+    name: guide-demo
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+      - name: Fetch kubectl
+        uses: azure/setup-kubectl@v3
+      - name: Fetch Teleport binaries
+        uses: teleport-actions/setup@v1
+        with:
+          version: (=teleport.version=)
+      - name: Fetch credentials using Machine ID
+        uses: teleport-actions/auth-k8s@v1
+        with:
+          # Use the address of the auth/proxy server for your own cluster.
+          proxy: example.domain:443
+          token: github-token
```

Comment:
Can we make it clear that this is the name of the Teleport join token and not of the GitHub token associated with the action run?

Reply:
afeaa0c40e